addons/project/tests/test_project_subtasks.py

```diff
@@ -242,15 +253,23 @@ def test_subtask_creation_with_form(self):
             4) check the correct nb of sub-tasks is displayed in the 'sub-tasks' stat button and on the parent task kanban card
             5) sub-tasks should be copied when the parent task is duplicated
         """
+        test_subtask_1 = self.env['project.task'].create({
+            'name': 'Test Subtask 1',
+        })
         with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
-            with task_form.child_ids.new() as subtask_form:
-                subtask_form.name = 'Test Subtask 1'
-                subtask_form.display_project_id = self.env['project.project']
+            task_form.child_ids.add(test_subtask_1)
+            task_form.child_ids[0].display_project_id = self.env['project.project']
+            task = task_form.save()
```

Comment:
If you need the result then I suggest to change the code like this:
```py
        task_form = Form(self.task_1.with_context({'tracking_disable': True}))
        task_form.child_ids.add(test_subtask_1)
        task_form.child_ids[0].display_project_id = self.env['project.project']
        task = task_form.save()
```

Reply:
I changed it, but what is the purpose of this change ? is it just some coding norm, or is it for a specific/functional reason ?

Reply:
Because after the `with` statement, you should not have the `task_form` (since it is defined in the `with` statement), it is allowed by Python language but it is a bit confusing, IMO. 🙂